### PR TITLE
Datacenters for asset tab

### PR DESCRIPTION
**Pull request by clairecfu**
- [x] linking curr selected datacenter to the asset table
- [x] displaying datacenter columns, sorting, filtering on datacenter name 
- [x] also, showing backend error message for unsuccessful login

### CODE REVIEW

File: rackcity/api/serializers/rack_serializers.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
 from rest_framework import serializers
 from rackcity.models import Rack
+from .datacenter_serializers import DatacenterSerializer
 
 
 class RackSerializer(serializers.ModelSerializer):
+    datacenter = DatacenterSerializer()
```
Comment (clairecfu) [inline rackcity/api/serializers/rack_serializers.py]: adding datacenter serializer to RackSerializer

File: src/utils/utils.tsx [hunk 2]
```diff
@@ -95,6 +96,11 @@ export interface RackRangeFields {
   num_end: number;
 }
 
+export interface SortFilterBody {
+  sort_by: Array<ITableSort>;
+  filters: Array<IFilter>;
+}
+
```
Comment (clairecfu) [inline src/utils/utils.tsx]: defining a new interface

File: src/utils/utils.tsx [hunk 3]
```diff
@@ -119,6 +125,7 @@ export interface CreateUserObject {
 }
 
 export interface RackObject extends ElementObject {
+  datacenter: DatacenterObject;
```
Comment (clairecfu) [inline src/utils/utils.tsx]: modifying RackObject to include DatacenterObject

File: src/components/elementView/elementTable.tsx [hunk 4]
```diff
@@ -499,13 +503,27 @@ class ElementTable extends React.Component<
...
+  // };
   setFieldNamesFromData = (items: Array<ElementObjectType>) => {
     let fields: Array<string> = [];
     Object.keys(items[0]).forEach((col: string) => {
       if (col === "model") {
         fields.push("model__vendor");
         fields.push("model__model_number");
+      } else if (col === "rack") {
+        fields.push("rack");
+        fields.push("rack__datacenter__name");
```
Comment (clairecfu) [inline src/components/elementView/elementTable.tsx]: adding new columns to table

File: src/components/elementView/elementTable.tsx [hunk 5]
```diff
@@ -843,9 +861,10 @@ class ElementTable extends React.Component<
                               <td>{value.model_number}</td>
                             ];
                           } else if (isRackObject(value)) {
-                            return (
-                              <td>{value.row_letter  + value.rack_num}</td>
-                            );
+                            return [
+                              <td>{value.row_letter + value.rack_num}</td>,
+                              <td>{value.datacenter.name}</td>
+                            ];
```
Comment (clairecfu) [inline src/components/elementView/elementTable.tsx]: showing the correct rack info

File: src/components/elementView/elementTable.tsx [hunk 6]
```diff
@@ -402,17 +406,17 @@ class ElementTable extends React.Component<
...
   updateSortData = (items: Array<ITableSort>) => {
-    const sorts_body = items.map(item => {
-      const { field, ascending } = item;
-      return { field, ascending };
-    });
-    console.log("detected new sorts ", sorts_body);
+    // const sorts_body = items.map(item => {
+    //   const { field, ascending } = item;
+    //   return { field, ascending };
+    // });
```
Comment (benauriemma) [inline src/components/elementView/elementTable.tsx]: Delete por favor

File: src/components/elementView/elementTable.tsx [hunk 7]
```diff
@@ -460,17 +464,17 @@ class ElementTable extends React.Component<
...
   }
   updateTableData = () => {
-    const sorts_body = this.state.sort_by.map(item => {
-      const { field, ascending } = item;
-      return { field, ascending };
-    });
+    // const sorts_body = this.state.sort_by.map(item => {
+    //   const { field, ascending } = item;
+    //   return { field, ascending };
+    // });
```
Comment (benauriemma) [inline src/components/elementView/elementTable.tsx]: ditto

File: src/components/elementView/elementTable.tsx [hunk 8]
```diff
@@ -499,13 +503,27 @@ class ElementTable extends React.Component<
...
+  //     if (field === "datacenter") {
+  //       return "rack__datacenter__name";
+  //     } else {
+  //       return field;
+  //     }
+  //   });
+
+  //   console.log(newFields);
+  //   return newFields;
+  // };
```
Comment (benauriemma) [inline src/components/elementView/elementTable.tsx]: Uno mas

File: src/components/elementView/elementTable.tsx (hunk 4 above)
Comment (benauriemma) [reply src/components/elementView/elementTable.tsx]: Sweeeet

**Pull request merged**